posix/posix.c

```diff
@@ -1207,20 +1374,25 @@ static int posix_fcntlDup(int fd, int fd2, int cloexec)
 	process_info_t *p;
 	int err;
 
-	if ((p = pinfo_find(proc_current()->process->id)) == NULL)
+	p = pinfo_find(proc_current()->process->id);
+	if (p == NULL) {
 		return -1;
+	}
 
 	proc_lockSet(&p->lock);
-	if (fd < 0 || fd > p->maxfd || fd2 < 0 || fd > p->maxfd) {
+	if ((fd < 0) || (fd > p->maxfd) || (fd2 < 0) || (fd2 > p->maxfd)) {
 		proc_lockClear(&p->lock);
 		pinfo_put(p);
 		return -EBADF;
 	}
 
-	while (p->fds[fd2].file != NULL && fd2++ < p->maxfd);
+	while ((p->fds[fd2].file != NULL) && (fd2++ < p->maxfd)) {
```

Comment:
i think that using increment in while condition is not compliant to misra rule 13.3, however I am not sure if I interpret that rule right

while loop may instead be changed to a for loop this way it will be compliant for sure and clearer imo